From e3cfbd049e2ac66fec5335ff24652fe291a7d973 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:10:31 -0400
Subject: [usb] Revert: pl2303: improve the chip type information output on startup

Message-id: <1410887463-20674-560-git-send-email-dzickus@redhat.com>
Patchwork-id: 95475
O-Subject: [RHEL7 PATCH 559/591] Revert "pl2303: improve the chip type information output on startup"
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit b52e111363e366202386f3e67f71681dbbb8e5d9
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Fri Nov 1 09:17:50 2013 -0700

    Revert "pl2303: improve the chip type information output on startup"

    This reverts commit a77a8c23e4db9fb1f776147eda0d85117359c700.

    Revert all of the pl2303 changes that went into 3.12-rc1 and -rc2 as
    they cause regressions on some versions of the chip.  This will all be
    revisited for later kernel versions when we can figure out how to handle
    this in a way that does not break working devices.

    Reported-by: Mika Westerberg <mika.westerberg@linux.intel.com>
    Cc: Frank Schafer <fschaefer.oss@googlemail.com>
    Acked-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/pl2303.c b/drivers/usb/serial/pl2303.c
index 8d4dafc..d4ca876 100644
--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -180,7 +180,6 @@ static int pl2303_startup(struct usb_serial *serial)
 {
  struct pl2303_serial_private *spriv;
  enum pl2303_type type = type_0;
- char *type_str = "unknown (treating as type_0)";
  unsigned char *buf;
 
  spriv = kzalloc(sizeof(*spriv), GFP_KERNEL);
@@ -193,18 +192,14 @@ static int pl2303_startup(struct usb_serial *serial)
   return -ENOMEM;
  }
 
- if (serial->dev->descriptor.bDeviceClass == 0x02) {
+ if (serial->dev->descriptor.bDeviceClass == 0x02)
   type = type_0;
-  type_str = "type_0";
- } else if (serial->dev->descriptor.bMaxPacketSize0 == 0x40) {
+ else if (serial->dev->descriptor.bMaxPacketSize0 == 0x40)
   type = HX;
-  type_str = "X/HX";
- } else if (serial->dev->descriptor.bDeviceClass == 0x00
-     || serial->dev->descriptor.bDeviceClass == 0xFF) {
+ else if (serial->dev->descriptor.bDeviceClass == 0x00
+   || serial->dev->descriptor.bDeviceClass == 0xFF)
   type = type_1;
-  type_str = "type_1";
- }
- dev_dbg(&serial->interface->dev, "device type: %s\n", type_str);
+ dev_dbg(&serial->interface->dev, "device type: %d\n", type);
 
  spriv->type = type;
  usb_set_serial_data(serial, spriv);
-- 
1.7.1